CMakeLists.txt

```diff
@@ -0,0 +1,201 @@
+# SPDX-FileCopyrightText: 2022 Dawid Wróbel <me@dawidwrobel.com>
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+cmake_minimum_required(VERSION 3.12)
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules)
+include(CMakeDependentOption)
+include(FeatureSummary)
+
+# set-up tool features; this needs to be done early to let the vcpkg auto-download all the optional dependencies
+option(ENABLE_ICONV "Build with ICONV" ON)
+add_feature_info("iconv" ENABLE_ICONV "Enables ICONV support for encoding conversion")
+if (ENABLE_ICONV)
+    list(APPEND VCPKG_MANIFEST_FEATURES "iconv")
+endif ()
+
+option(ENABLE_OFXCONNECT "Enable ofxconnect tool" ON)
+add_feature_info("ofxconnect" ENABLE_OFXCONNECT "Enables ofxconnect utility which allows to test OFX Direct Connect")
+if (ENABLE_OFXCONNECT)
+    list(APPEND VCPKG_MANIFEST_FEATURES "ofxconnect")
+endif ()
+
+option(ENABLE_OFXDUMP "Enable ofxdump tool" ON)
+add_feature_info("ofxdump" ENABLE_OFXDUMP "Enables ofxdump utility which prints, in human readable form, everything the library understands about a file")
+if (ENABLE_OFXDUMP)
+    list(APPEND VCPKG_MANIFEST_FEATURES "ofxdump")
+endif ()
+
+option(ENABLE_OFX2QIF "Enable ofx2qif tool" ON)
+add_feature_info("ofx2qif" ENABLE_OFX2QIF "Enables OFX file to QIF (Quicken Interchange Format) file converter")
+if (ENABLE_OFX2QIF)
+    list(APPEND VCPKG_MANIFEST_FEATURES "ofx2qif")
+endif ()
+
+# actually set up the project
+project(libofx)
+
+set(LIBOFX_MAJOR_VERSION 0)
+set(LIBOFX_MINOR_VERSION 10)
+set(LIBOFX_MICRO_VERSION 5)
+set(LIBOFX_VERSION_RELEASE_STRING ${LIBOFX_MAJOR_VERSION}.${LIBOFX_MINOR_VERSION}.${LIBOFX_MICRO_VERSION})
+
+# If no build type is set, use "Release with Debug Info"
+if (NOT CMAKE_BUILD_TYPE)
+    set(CMAKE_BUILD_TYPE RelWithDebInfo)
+endif (NOT CMAKE_BUILD_TYPE)
+
+# locate required dependencies
+find_package(OpenSP REQUIRED)
+
+# locate optional dependencies
+if (ENABLE_OFXCONNECT)
+    find_package(CURL REQUIRED)
+    if (CURL_FOUND)
+        set(HAVE_LIBCURL 1)
+    endif ()
+
+    find_package(PkgConfig)
+    if (PkgConfig_FOUND)
+        pkg_check_modules(LIBXMLPP IMPORTED_TARGET libxml++-2.6>=2.6)
```

Comment:
> One thing that's missing is a dist target: autotools provides one by default, cmake doesn't. The main purpose is to have a working configure script that doesn't need autotools installed. 

Why would you insist on generating that tarball with CMake instead of autotools, then? Especially that it was agreed that support for autotools (and a pre-generated configure script) would be dropped around +1 release, so all this effort would be futile.

> The other reason is to have a tarball that's named the way the FOSS world expects instead of Github's version-only vX.Y.tar.gz with no project name.

I second my opinion that it seems unnecessary. Outside of your quoted example of Gnucash, I have not once seen a software that did that just for that purpose. 

But most importantly, pretty much all of the software distribution systems I am aware of can work with Git directly, either via a tag, branch or SHA, and don't need a tarball anymore. I'd argue that a tarball generated by Github would suffice.

Reply:
> needs
>
> if (NOT LIBXMLPP_FOUND)
>    set(ENABLE_OFXCONNECT OFF)
> endif()

I strongly disagree on that. If `ENABLE_OFXCONNECT` is enabled and a required dependency is not found, then it should fail as it does instead of transparently changing a user-facing feature. Such implicit behavior is easy to miss, especially when used in automated builds. I can easily imagine `libxml` somehow getting removed from the underlying CI/CD system, resulting in a successful build with a falsely-advertised OFXCONNECT functionality.

Reply:
> But most importantly, pretty much all of the software distribution systems I am aware of can work with Git directly, either via a tag or branch, accompanied by a SHA, and don't need a hosted tarball anymore. In any other case, I'd argue that a tarball generated by Github would suffice. Adding extra scripting sounds very un-K.I.S.S. to me.

OK, but it's a change in behavior and means that @benoitg and @cstim have to change their release process. Whether they want to do that is up to them.

>  If ENABLE_OFXCONNECT is enabled and a required dependency is not found, then it should fail as it does instead of transparently changing a user-facing configurable feature. 

That's a change in behavior, configure.ac does:
```
PKG_CHECK_MODULES(LIBCURL, libcurl >= 7.9.7,
        [libcurl_available=yes],
        [libcurl_available=no
         AC_MSG_WARN([libcurl is not available. ofxconnect (Direct connect samples) will NOT be built.])
        ])

PKG_CHECK_MODULES(LIBXMLPP,libxml++-2.6 >= 2.6,
	[
	AC_DEFINE(HAVE_LIBXMLPP, 1, [Defined if libxml++ is available])
	have_libxmlpp=yes],
        [AC_MSG_WARN([libxml++ 2.6 is not available. ofxconnect (Direct connect samples) will NOT be built.])
	have_libxmlpp=no])
```

A reasonable alternative would be to default `ENABLE_OFXCONNECT` to `OFF` so that a default build doesn't fail for an optional component. You could also provide a more verbose explanatory error message suggesting that the failure is because of an optional component that's enabled by default and that the builder can disable it instead of installing the dependencies.

A more serious concern and a bit off topic: Does ofxconnect still work? ofxpartner.cpp has a hard-coded url `http://moneycentral.msn.com/money/2005`. `moneycentral.msn.com` isn't in the DNS anymore. That aside, GnuCash uses [AQBanking](https://www.aquamaniac.de) for OFX-DirectConnect support and I've worked with its developer several times in the last few years to deal with changes Intuit has made to the way the banks' servers authenticate. The [last functional commit](https://github.com/libofx/libofx/commit/74e8e94e1abb2bb316239e410b33822bb4d91fa1) to ofxconnect was in 2007,  more than 15 years ago. Everything since has been just to keep it building.

Reply:
> OK, but it's a change in behavior and means that @benoitg and @cstim have to change their release process. 

Since the tarballs are already published as is on Github already — albeit without the autootool scripting — hopefully it won't be much of a change once autotool scripting gets dropped. 

> That's a change in behavior, configure.ac does
> A reasonable alternative would be to default ENABLE_OFXCONNECT 

Building of the tools wasn't even configurable until very recently (see: https://github.com/libofx/libofx/commit/f4f7c473beae782e15bbd3fdad1fd4543dd8c3ff), so the behavior of `./configure` script that you refer to seems reasonable. All the tools are configurable separately in CMake, though, so it isn't much of an issue as they can be disabled at will when dependencies are missing.

> A reasonable alternative would be to default ENABLE_OFXCONNECT to OFF

That's what I initially proposed, but @cstim suggested otherwise: https://github.com/libofx/libofx/pull/63#pullrequestreview-1048681174

> You could also provide a more verbose explanatory error message suggesting that the failure is because of an optional component that's enabled by default and that the builder can disable it instead of installing the dependencies.

Actually, I thought that `feature_summary(WHAT REQUIRED_PACKAGES_NOT_FOUND` would make it obvious enough as to what caused what, without custom messaging. But it turns out that it's not even shown when first missing but otherwise required package is found, and that's a bit puzzling.  Trying to figure it out.

And there's that, too: https://gitlab.kitware.com/cmake/cmake/-/issues/22689

> A more serious concern and a bit off topic: Does ofxconnect still work? 

Actually I was wondering about the same — see the `Is ofxconnect's test-privateserver.sh still functional?` in my first comment that touches on that.

Reply:
>Is ofxconnect's test-privateserver.sh still functional?

It [requires a gpg key](https://github.com/libofx/libofx/blob/master/ofxconnect/README.privateserver) to test it or even to see what it does. I'll try the email address in that readme, but I don't have much hope. But it's also orthogonal to the real problem because even if that server is still active it isn't running what real banks do. If that private server is still up it probably works with ofxconnect, but that provides no indication of whether ofxconnect will connect to a real bank.

Reply:
> Actually, I thought that feature_summary(WHAT REQUIRED_PACKAGES_NOT_FOUND would make it obvious enough as to what caused what, without custom messaging. But it turns out that it's not even shown, since when first missing but otherwise required package is found, CMake fails on the spot. Very puzzling, trying to figure it out.

OK, I sorted that all out with 7a54725f8276d49d06a6a08037d21b25a0c69113, which shows a meaningful error when a package is missing, alongside the feature summary. It doesn't work for `pkg_check_modules`, though.

Reply:
> It [requires a gpg key](https://github.com/libofx/libofx/blob/master/ofxconnect/README.privateserver) to test it or even to see what it does. I'll try the email address in that readme, but I don't have much hope. But it's also orthogonal to the real problem because even if that server is still active it isn't running what real banks do. If that private server is still up it probably works with ofxconnect, but that provides no indication of whether ofxconnect will connect to a real bank.

I suppose only  @benoitg and @cstim  can answer that

Reply:
> > OK, but it's a change in behavior and means that @benoitg and @cstim have to change their release process.
> 
> Since the tarballs get published as is on Github already — albeit without the autootool scripting — hopefully it won't be much of a change once autotool scripting gets dropped.

I really don't mind.  Your work to modernize the build infrastructure to improve cross-platform building is a large piece of work.  Tweaking the (currently manual) release process is comparatively trivial.